code/datums/weather/weather_types/acid_rain.dm

```diff
@@ -64,7 +64,23 @@
 	end_message = span_boldannounce("The downpour gradually slows to a light shower.")
 	end_overlay = "rain_low"
 
-	aesthetic = TRUE
-
 	probability = 60
 	repeatable = TRUE
+
+/datum/weather/acid_rain/harmless/weather_act(mob/living/carbon/human/L)
+	cleanup(L)
+
+/datum/weather/acid_rain/proc/cleanup(mob/living/carbon/human/H)
+	if(!ishuman(H))
+		return
+	if(H.back?.clean_blood())
+		H.update_inv_back()
+	if(H.wear_suit?.clean_blood())
+		H.update_inv_wear_suit()
+	if(H.w_uniform?.clean_blood())
+		H.update_inv_w_uniform()
+	if(H.gloves?.clean_blood())
+		H.update_inv_gloves()
+	if(H.shoes?.clean_blood())
+		H.update_inv_shoes()
+	H.clean_blood()
```

Comment:
```suggestion
	if(!ishuman(L))
		return
	var/mob/living/carbon/human/H = L
	if(H.back?.clean_blood())
		H.update_inv_back()
	if(H.wear_suit?.clean_blood())
		H.update_inv_wear_suit()
	if(H.w_uniform?.clean_blood())
		H.update_inv_w_uniform()
	if(H.gloves?.clean_blood())
		H.update_inv_gloves()
	if(H.shoes?.clean_blood())
		H.update_inv_shoes()
	H.clean_blood()
```

Reply:
Shower has a fairly detailed proc to cover washing a mob. Would it make more sense to make a wash proc at the carbon level, then some extra stuff at human etc, and have shower/rain just call that?